packages/viewer/src/modules/extensions/measurements/MeasurementsExtension.ts

```diff
@@ -423,6 +437,36 @@ export class MeasurementsExtension extends Extension {
     }
   }
 
+  /**
+   * Idempotent way of setting measurements
+   */
+  public setMeasurements(measurements: MeasurementData[]) {
```

Comment:
cc @AlexandruPopovici these changes seem to work

ultimately we need an idempotent setMeasurements() function that will take care of removing unnecessary ones and adding necessary ones, and do nothing if nothing needs to be done